### PR TITLE
Add deserialization

### PR DESCRIPTION
**Pull request by tekknolagi**

### CODE REVIEW

File: scrapscript.py [hunk 1]
```diff
@@ -2630,6 +2869,149 @@ def test_serialize_record(self) -> None:
...
+            "right": {"type": "Int", "value": 456},
+            "type": "Binop",
+        }
+        obj = Binop(BinopKind.ADD, Int(123), Int(456))
+        self.assertEqual(Object.deserialize(msg), obj)
+
+    def test_deserialize_env_object(self) -> None:
+        obj = EnvObject({"x": Int(1)})
+        msg = {"env": {"x": {"type": "Int", "value": 1}}, "type": "EnvObject"}
+        self.assertEqual(Object.deserialize(msg), obj)
```
Comment (surprisetalk) [inline scrapscript.py:2967]: I hadn't thought of deserializing the env object, but I like the idea the more I think of it. Great work

**Review by surprisetalk — APPROVED** [review]
Excellent readable code, as always! The tests have been providing an excellent entry point for how particular things are implemented.

Unsure if it's possible, but we may be able to implement bool as variants in the future, if that makes things simpler: `# true () # false ()`

**Comment by surprisetalk** [comment]
dang missed your merge by 1 minute haha

**Comment by tekknolagi** [comment]
> Unsure if it's possible, but we may be able to implement bool as variants in the future, if that makes things simpler: `# true () # false ()`

Yep, that is the goal. We just don't have variants yet :)

**Pull request merged**